clients/src/main/java/org/apache/kafka/common/utils/ExponentialBackoff.java

```diff
@@ -32,7 +32,16 @@
     private final double expMax;
     private final long initialInterval;
     private final double jitter;
+    private long attemptedCount = 0;
```

Comment:
add the `attemptedCount` in `ExponentialBackoff` class. The caller can use the `attemptedCount` and doesn't need to maintain the attempted count in their side. It's good when the `ExponentialBackoff` only has single place to backoff, or the caller is inside lambda expression.

Reply:
Is it guaranteed instances of this class are confined to one thread hence no synchronization is needed?